rasterio/_warp.pyx

```diff
@@ -487,3 +491,60 @@ def _reproject(
         if dtypes.is_ndarray(source):
             if hdsin != NULL:
                 _gdal.GDALClose(hdsin)
+
+
+def _calculate_default_transform(
+        src_crs, dst_crs, width, height, left, bottom, right, top,
+        resolution=None, densify_pts=21):
+    """Wraps GDAL's algorithm."""
+
+    cdef void *hTransformArg = NULL
+    cdef int npixels = 0
+    cdef int nlines = 0
+    cdef double extent[4]
+    cdef double geotransform[6]
+    cdef void *osr = NULL
+    cdef char *wkt = NULL
+    cdef InMemoryRaster temp = None
+
+    extent[:] = [0.0, 0.0, 0.0, 0.0]
+    geotransform[:] = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+
+    # Make an in-memory raster dataset we can pass to 
+    # GDALCreateGenImgProjTransformer().
+    transform = from_bounds(left, bottom, right, top, width, height)
+    img = np.ones((height, width))
+
+    osr = _base._osr_from_crs(dst_crs)
+    _gdal.OSRExportToWkt(osr, &wkt)
+    _gdal.OSRDestroySpatialReference(osr)
+
+    with InMemoryRaster(
+            img, transform=transform.to_gdal(), crs=src_crs) as temp:
+        hTransformArg = _gdal.GDALCreateGenImgProjTransformer(
```

Comment:
@brendan-ward They both do the same thing but It looks like [GDALCreateGenImgProjTransformer2](http://www.gdal.org/gdal__alg_8h.html#a94cd172f78dbc41d6f407d662914f2e3]) just takes options which provide a lot more fine-grained control. In particular, it looks like Transformer2 is required if we want to pursue GCPs, #203. If we need or anticipate needing any of those options, we should move to GDALCreateGenImgProjTransformer2 but Transformer does the same job for now.